File: data/code_pairs/fortran/2349.f90
module NeighborFinder
contains
    subroutine FindElemEdgeNeighbor(ElemConnectivity, ElemId1, ElemEdge1, bflag, ElemId2, ElemEdge2)
        integer, intent(in) :: ElemConnectivity(:,:)
        integer, intent(in) :: ElemId1, ElemEdge1
        logical, intent(out) :: bflag
        integer, intent(out) :: ElemId2, ElemEdge2

        bflag = .false.
        ElemId2 = ElemId1 + 1
        ElemEdge2 = ElemEdge1 + 1
    end subroutine FindElemEdgeNeighbor

    subroutine FindElemEdgeNeighbor2(ElemConnectivity, ElemId1, ElemEdge1, bflag, ElemId2, ElemEdge2)
        integer, intent(in) :: ElemConnectivity(:,:)
        integer, intent(in) :: ElemId1, ElemEdge1
        logical, intent(out) :: bflag
        integer, intent(out) :: ElemId2, ElemEdge2

        bflag = .true.
        ElemId2 = ElemId1 - 1
        ElemEdge2 = ElemEdge1 - 1
    end subroutine FindElemEdgeNeighbor2
end module NeighborFinder

File: data/code_pairs/cpp/2349.cpp
#include <iostream>
#include <vector>

// Declaration of functions
void FindElemEdgeNeighbor(const std::vector<std::vector<int>>& ElemConnectivity, 
                          int ElemId1, int ElemEdge1, bool& bflag, 
                          int& ElemId2, int& ElemEdge2) {
    // Simplified logic for demonstration
    bflag = false;
    ElemId2 = ElemId1 + 1;
    ElemEdge2 = ElemEdge1 + 1;
}

void FindElemEdgeNeighbor2(const std::vector<std::vector<int>>& ElemConnectivity, 
                           int ElemId1, int ElemEdge1, bool& bflag, 
                           int& ElemId2, int& ElemEdge2) {
    // Simplified logic for demonstration
    bflag = true;
    ElemId2 = ElemId1 - 1;
    ElemEdge2 = ElemEdge1 - 1;
}

// Main function including basic tests
int main() {
    std::vector<std::vector<int>> ElemConnectivity = {{1, 2}, {3, 4}};
    int ElemId1 = 1, ElemEdge1 = 1;
    int ElemId2, ElemEdge2;
    bool bflag;

    FindElemEdgeNeighbor(ElemConnectivity, ElemId1, ElemEdge1, bflag, ElemId2, ElemEdge2);
    std::cout << "Test 1 - bflag: " << bflag << ", ElemId2: " << ElemId2 << ", ElemEdge2: " << ElemEdge2 << std::endl;

    FindElemEdgeNeighbor2(ElemConnectivity, ElemId1, ElemEdge1, bflag, ElemId2, ElemEdge2);
    std::cout << "Test 2 - bflag: " << bflag << ", ElemId2: " << ElemId2 << ", ElemEdge2: " << ElemEdge2 << std::endl;

    return 0;
}